### ISSUE TITLE
Input contains NaN, infinity or a value too large for dtype('float32').

### ISSUE THREAD

**Issue by similang** [pycaret/pycaret#290]
Encountered this error when running `compare_models` step. However, the data does not have missing value or inf. 

`print(np.any(np.isnan(X_train)),np.all(np.isfinite(X_train)))` returns False, True

Not sure what's the issue with my data.

**Comment by GLevV** [comment]
Can you please elaborate? At least add the whole error output
The problem could be not in the input data but in predictions, when they fed to function that calculates metric.

**Comment by similang** [comment]
Here's the full error output.

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-29-9d46714d5836> in <module>
----> 1 compare_models(sort='AUC',fold=10)

d:\anaconda\envs\pyc\lib\site-packages\pycaret\classification.py in compare_models(blacklist, fold, round, sort, turbo)
   3434             if hasattr(model, 'predict_proba'):
   3435 
-> 3436                 model.fit(Xtrain,ytrain)
   3437                 pred_prob = model.predict_proba(Xtest)
   3438                 pred_prob = pred_prob[:,1]

d:\anaconda\envs\pyc\lib\site-packages\sklearn\tree\_classes.py in fit(self, X, y, sample_weight, check_input, X_idx_sorted)
    874             sample_weight=sample_weight,
    875             check_input=check_input,
--> 876             X_idx_sorted=X_idx_sorted)
    877         return self
    878 

d:\anaconda\envs\pyc\lib\site-packages\sklearn\tree\_classes.py in fit(self, X, y, sample_weight, check_input, X_idx_sorted)
    147 
    148         if check_input:
--> 149             X = check_array(X, dtype=DTYPE, accept_sparse="csc")
    150             y = check_array(y, ensure_2d=False, dtype=None)
    151             if issparse(X):

d:\anaconda\envs\pyc\lib\site-packages\sklearn\utils\validation.py in check_array(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, warn_on_dtype, estimator)
    560         if force_all_finite:
    561             _assert_all_finite(array,
--> 562                                allow_nan=force_all_finite == 'allow-nan')
    563 
    564     if ensure_min_samples > 0:

d:\anaconda\envs\pyc\lib\site-packages\sklearn\utils\validation.py in _assert_all_finite(X, allow_nan, msg_dtype)
     58                     msg_err.format
     59                     (type_err,
---> 60                      msg_dtype if msg_dtype is not None else X.dtype)
     61             )
     62     # for object dtype data, we only check for NaNs (GH-13254)

ValueError: Input contains NaN, infinity or a value too large for dtype('float32').

**Comment by similang** [comment]
Not sure if the above error is helpful.. if needed I can share with you the data. Thanks.

**Comment by GLevV** [comment]
Can you run this on your training data, just to be sure:
`X_train = X_train.replace((np.inf, -np.inf, np.nan), 0).reset_index(drop=True)`
Problem seems to occur in DecisionTreeClassifier input check, but it is not the first model in the default compare_model queue. Can you try to add it to blacklist, like this
`compare_model(blacklist=[dt])`

**Comment by similang** [comment]
Hi, same error with the following code:

`
compare_models(sort='AUC',fold=10, blacklist=['dt'])
`
![image](https://user-images.githubusercontent.com/7208012/87938803-af822280-cac9-11ea-96ac-f80c2003f5e2.png)

```
--------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-22-c439c54e7e1d> in <module>
----> 1 compare_models(sort='AUC',fold=10, blacklist=['dt'])

d:\anaconda\envs\pyc\lib\site-packages\pycaret\classification.py in compare_models(blacklist, fold, round, sort, turbo)
   3434             if hasattr(model, 'predict_proba'):
   3435 
-> 3436                 model.fit(Xtrain,ytrain)
   3437                 pred_prob = model.predict_proba(Xtest)
   3438                 pred_prob = pred_prob[:,1]

d:\anaconda\envs\pyc\lib\site-packages\sklearn\ensemble\_forest.py in fit(self, X, y, sample_weight)
    293         """
    294         # Validate or convert input data
--> 295         X = check_array(X, accept_sparse="csc", dtype=DTYPE)
    296         y = check_array(y, accept_sparse='csc', ensure_2d=False, dtype=None)
    297         if sample_weight is not None:

d:\anaconda\envs\pyc\lib\site-packages\sklearn\utils\validation.py in check_array(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, warn_on_dtype, estimator)
    560         if force_all_finite:
    561             _assert_all_finite(array,
--> 562                                allow_nan=force_all_finite == 'allow-nan')
    563 
    564     if ensure_min_samples > 0:

d:\anaconda\envs\pyc\lib\site-packages\sklearn\utils\validation.py in _assert_all_finite(X, allow_nan, msg_dtype)
     58                     msg_err.format
     59                     (type_err,
---> 60                      msg_dtype if msg_dtype is not None else X.dtype)
     61             )
     62     # for object dtype data, we only check for NaNs (GH-13254)

ValueError: Input contains NaN, infinity or a value too large for dtype('float32').
```

**Comment by GLevV** [comment]
Now it produces error in RandomForestClassifier, which is just a bunch of decision trees that you've blacklisted. Somewhy your dataset is acceptable by other classifiers and is not for tree-based methods. 
It is really strange, you can try to blacklist all the tree based models (['dt', 'rf', 'et', 'ada', 'gbc']), but this is palliative. Maybe some error in preprocessing, when you setup pycaret environment.

**Comment by pycaret** [comment]
@similang Can you please provide the sample dataset so that I can try to reproduce the issue on my end.

Also, can you please let me know the version of pandas and scikit-learn.

Thanks.

**Comment by similang** [comment]
Files are big: 

X_train: https://drive.google.com/file/d/1EXyVnpMpC7DgrNV88YbfCCUf8gA7ZAD5/view?usp=sharing
y_train: https://drive.google.com/file/d/1qXJ-b0YKQQ06JC7FO4SdNy_7hm7GgVJ7/view?usp=sharing

If too large to download, I will redo - with some sub-sampling - as I don't know which rows are with mistakes. 

Also on the version: 
![image](https://user-images.githubusercontent.com/7208012/88087695-e33d7500-cbbb-11ea-8554-d05d470a8966.png)

**Comment by pycaret** [comment]
This issue will be fixed in `2.2` release expected by the end of Sep.

**Comment by pycaret** [comment]
@similang 

This issue has been fixed in the `pycaret-nightly` release. Please install pycaret-nightly using `pip install pycaret-nightly`. Feel free to open a new issue, in case the problem persists. This fix will be published to PyPi as `pycaret==2.2.0` on October 28, 2020.
